cirq/sim/state.py

```diff
@@ -121,3 +123,157 @@ def validate_normalized_state(state: np.ndarray, num_qubits: int,
     norm = np.sum(np.abs(state) ** 2)
     if not np.isclose(norm, 1):
         raise ValueError('State is not normalized instead had norm %s' % norm)
+
+
+def sample_state_vector(
+    state: np.ndarray,
+    indices: List[int],
+    repetitions: int=1) -> List[List[bool]]:
+    """Samples repeatedly from measurements in the computational basis.
+
+    Note that this does not modify the passed in state.
+
+    Args:
+        state: The state to be measured. This state is assumed to be normalized.
+            The state must be of size 2 ** integer.  The state can be of shape
+            (2 ** integer) or (2, 2, ..., 2).
+        indices: Which qubits are measured. The state is assumed to be supplied
+            in big endian order. That is the xth index of v, when expressed as
+            a bitstring, has the largest values that the 0th index.
+        repetitions: The number of times to sample the state.
+
+    Returns:
+        Measurement results with True corresponding to the |1> state.
+        The outer list is for repetitions, and the inner corresponds to
+        measurements ordered by the input indices.
+
+    Raises:
+        ValueError if repetitions is less than one or size of state is not a
+            power of 2.
+        IndexError if the indices are out of range for the number of qubits
+            corresponding to the state.
+    """
+    if repetitions < 1:
+        raise ValueError(
+            'Number of repetitions cannot be negative. Was {}'.format(
+                repetitions))
+    num_qubits = _validate_num_qubits(state)
+    _validate_indices(num_qubits, indices)
+
+    if len(indices) == 0:
+        return [[]]
```

Comment:
This result doesn't satisfy the invariant that the length of the other list is equal to the number of repetitions. Add a test that `sample_state_vector(..., indices=[], repetitions=2)` returns `[[], []]`.

If you return a list of tuples, instead of a list of lists, you can return `[()] * repetitions` instead of `[[] for _ in range(repetitions)]` (need to avoid aliasing between the items).

Reply:
Done but note that this is in conflict with your desire to support zero repetitions.

Reply:
How is it in conflict? `[[] for _ in range(0)]` returns `[]`, which is the correct result for 0 repetitions.

Reply:
ah true!